### PR TITLE
mgmt, redis update default version to V6

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Fix https://github.com/Azure/azure-sdk-for-java/issues/31232

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/RedisCacheImpl.java [hunk 1]
```diff
@@ -586,6 +587,9 @@ public Mono<RedisCache> updateResourceAsync() {
     public Mono<RedisCache> createResourceAsync() {
         createParameters.withLocation(this.regionName());
         createParameters.withTags(this.innerModel().tags());
+        if (CoreUtils.isNullOrEmpty(this.createParameters.redisVersion())) {
+            withRedisVersion(RedisVersion.V6);
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/RedisCacheImpl.java:591]: which version does RedisVersion.V6 refer? 6.0.0?

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/RedisCacheImpl.java:591]: It's major redis version(e.g. 4, 6), not a specific(minor, e.g. 6.0.0) one. Service only allow configuring with major version, and in backend, they will choose a minor version(I assume the latest).

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java [hunk 2]
```diff
@@ -581,6 +585,23 @@ enum RedisVersion {
...
+         * Gets the major version from the given redis version.
+         * @param redisVersion valid redis version, e.g. 6.0.14
+         * @return the major redis version, or null if the given redisVersion is invalid
+         */
+        public static RedisVersion fromVersion(String redisVersion) {
+            if (CoreUtils.isNullOrEmpty(redisVersion)) {
+                return null;
+            }
+            for (RedisVersion version : values()) {
+                if (redisVersion.startsWith(version.getValue())) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java:599]: nit on startWith... when redis get to 41.0.0

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java [hunk 3]
```diff
@@ -581,6 +585,23 @@ enum RedisVersion {
         public String getValue() {
             return value;
         }
+
+        /**
+         * Gets the major version from the given redis version.
+         * @param redisVersion valid redis version, e.g. 6.0.14
+         * @return the major redis version, or null if the given redisVersion is invalid
+         */
+        public static RedisVersion fromVersion(String redisVersion) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java:594]: If it is only used in test, should we add a public method here?

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java:599]: Right... Thanks...

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java (hunk 3 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java:594]: Our `RedisCache.redisVersion()` returns minor version string, e.g. 6.x.x. Maybe we can add a convenience method for converting minor version to major version?

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java [hunk 4]
```diff
@@ -564,7 +565,10 @@ enum RedisVersion {
         V6("6"),
         /**
          * version 4.x.x
+         * @deprecated Because Redis version 4 is no longer supported by the open source community,
+         *             it will be retired from Azure Cache for Redis.
          */
+        @Deprecated
         V4("4");
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java:572]: A warning here, adding value to enum when it is used only in input is fine. But adding value when the enum is used in output is considered breaking change.

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java:572]: Got it. The extra method will introduce breaking change when new versions are introduced. Will remove it.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**